cmake/modules/BuildArrow.cmake

```diff
@@ -16,7 +16,7 @@ function(build_arrow)
   list(APPEND arrow_CMAKE_ARGS -DARROW_JEMALLOC=OFF)
 
   # transitive dependencies
-  list(APPEND arrow_INTERFACE_LINK_LIBRARIES thrift)
+  list(APPEND arrow_INTERFACE_LINK_LIBRARIES thrift::libthrift)
```

Comment:
@leonid-s-usov Hi Leonid, if you sure that some of these changes are correct and not specific to MacOS, i'd suggest send them separately in different PR. we can and should review and merge them separately.

Reply:
a changeset with 23 commits would scare most reviewers. and this specific CMake change involves lots of different kinds of changes span multiple areas. so it should be split into different commits and even different PRs.

Reply:
I'm afraid this will make things even harder. I have several PRs for some of the commits from here which make sense standalone. Some even got merged, but here's one that's waiting forever https://github.com/ceph/ceph/pull/54944


All of the changes in this PR are needed to build on Mac OS, which is why it makes sense to combine them. If I pick, let's say, this change, then it can't be validated as solving a Mac issue because other changes won't be included, and for non-Mac environments, this change should be a Noop.

Another issue, there are so many seemingly unrelated changes that it will take me several days of work to open all those PRs. Without any guarantees that it would make it faster I am not keen to begin this rather tedious task.

This PR is passing builds as you can see with the CI. It has also been validated by up to 10 Mac users. The main blocker for this PR was several submodule changes which I did open up as separate PRs and spent 6 months pinging people to have those merged.

As I see it, all we need now is confirmation from a couple of non-Mac users that taking this as a whole doesn't break their environment, and a generic teuthology run to verify this doesn't introduce bugs.

May I ask you to merge it into your branch and make a build just as a data point?

Reply:
@tchaikov do you think it would facilitate things if I created a commit per component instead of a commit per topic?

Reply:
> Another issue, there are so many seemingly unrelated changes that it will take me several days of work to open all those PRs. Without any guarantees that it would make it faster I am not keen to begin this rather tedious task.

i cannot guarantee anything. but i think you could start by extracting smaller ones into separate PRs.

Reply:
> @tchaikov do you think it would facilitate things if I created a commit per component instead of a commit per topic?

smaller commits **and** smaller PRs . 100 commits in a PR is still not the way to go.

Reply:
> a changeset with 23 commits

there are 13 commits in this PR. I'll see how I can make it easier for the reviewers. Maybe I'll get help from other people interested so that I don't have to generate all those PRs alone